orbit/cmd/orbit/orbit.go

```diff
@@ -457,8 +468,33 @@ func main() {
 					log.Info().Msg("No cert chain available. Relying on system store.")
 				}
 			}
+
 		}
 
+		orbitClient, err := service.NewOrbitClient(fleetURL, c.String("fleet-certificate"), c.Bool("insecure"))
+		if err != nil {
+			return fmt.Errorf("error new orbit client: %w", err)
+		}
+		orbitNodeKey, err := getOrbitNodeKeyOrEnroll(orbitClient, c.String("root-dir"), enrollSecret, uuidStr)
```

Comment:
New orbit with an old version of Fleet won't work because it expects the Enroll API.

This whole new process (of enrolling Orbit to Fleet) should be run only if the server supports it.
If the enrolling API returns a 404 then none of this new functionality should be executed.

Reply:
@roperzh I think we should tackle this on your PR as part of https://github.com/fleetdm/fleet/pull/7737#discussion_r970694876. (So, Sharvil doesn't need to tackle this comment on this PR.)

Makes sense?

Reply:
just to over-communicate what we discussed elsewhere: makes sense 👍